src/Sparrow/Utils/TimeoutManager.cs

```diff
@@ -14,10 +16,20 @@ namespace Sparrow.Utils
 {
     internal static class TimeoutManager
     {
+        private static FrozenDictionary<uint, TimerTaskHolder> ValuesForRead = new Dictionary<uint, TimerTaskHolder>().ToFrozenDictionary();
         private static readonly ConcurrentDictionary<uint, TimerTaskHolder> Values = new ConcurrentDictionary<uint, TimerTaskHolder>();
         private static readonly Task InfiniteTask = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously).Task;
 
-        private sealed class TimerTaskHolder  : IDisposable
+        private const bool ForceTaskDelay = true;
+
+        private static readonly bool UseTaskDelay;
+
+        static TimeoutManager()
+        {
+            UseTaskDelay = Environment.ProcessorCount <= 2 || ForceTaskDelay;
```

Comment:
maybe env instead?

Reply:
Would probably be better to have that as env yes.

Reply:
We want that only on burstable, basically.